core/domain/exp_domain.py

```diff
@@ -3060,90 +3171,1460 @@ def _convert_states_v51_dict_to_v52_dict(
+                None, removes the tag
+                - If `skill_id-with-value` attribute is not present or empty or
+                None, removes the tag
+            - `oppia-noninteractive-math`
+                - If `math_content-with-value` attribute not in math tag,
+                removes the tag
+                - If `raw_latex` is not present or empty or None, removes
+                the tag
+            - `oppia-noninteractive-video`
+                - If `start-with-value` or `end-with-value` is not present,
+                introduce them to the tag and assign 0 to them
+                - If `autoplay-with-value` is not present or is not boolean,
+                introduce it to the tag and assign `false` to them
+                - If `video_id-with-value` is not present or empty, removes
+                the tag
+                - If `start-with-value` > `end-with-value`, set both to '0'
+            - `oppia-noninteractive-link`
+                - If `text-with-value` or `url-with-value` is not present,
+                or is empty simply removes the tag
+            - `oppia-noninteractive-tabs` and `oppia-noninteractive-collapsible`
+                - If these tags are nested inside tabs and collapsible tag, we
+                will simply remove the tag
+
+        Args:
+            html: str. The RTE tags.
+            is_tags_nested_inside_tabs_or_collapsible: bool. If the tag is
+                present inside the tabs or collapsible tag.
+
+        Returns:
+            str. Returns the updated html value.
+        """
+        empty_values = ['&quot;&quot;', '', '\'\'', '\"\"']
+        soup = bs4.BeautifulSoup(html, 'html.parser')
+
+        for tag in soup.find_all('oppia-noninteractive-image'):
+            if not tag.has_attr('alt-with-value'):
+                tag['alt-with-value'] = '&quot;&quot;'
+
+            if not tag.has_attr('filepath-with-value'):
+                tag.decompose()
+                continue
+
+            if tag['filepath-with-value'] in empty_values:
+                tag.decompose()
+                continue
+
+            if not tag.has_attr('caption-with-value'):
+                tag['caption-with-value'] = '&quot;&quot;'
+
+        for tag in soup.find_all('oppia-noninteractive-skillreview'):
+            if not tag.has_attr('text-with-value'):
+                tag.decompose()
+                continue
+
+            if tag['text-with-value'] is None:
+                tag.decompose()
+                continue
+            if tag['text-with-value'].strip() in empty_values:
+                tag.decompose()
+                continue
+
+            if not tag.has_attr('skill_id-with-value'):
+                tag.decompose()
+                continue
+
+            if tag['skill_id-with-value'] is None:
+                tag.decompose()
+                continue
+            if tag['skill_id-with-value'].strip() in empty_values:
+                tag.decompose()
+                continue
```

Comment:
```suggestion
            if (
                not tag.has_attr('text-with-value') or
                tag['text-with-value'] is None or
                tag['text-with-value'].strip() in empty_values or
                not tag.has_attr('skill_id-with-value') or
                tag['skill_id-with-value'] is None or
                tag['skill_id-with-value'].strip() in empty_values
            ):
                tag.decompose()
                continue
```

Or maybe you can parameterized the three operations that you use (`tag.has_attr(x)`, `tag[x] is None`, `tag[x].strip() in empty_values`) and separate them into a function and the call the function for each param.

Reply:
Done